components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/SessionMgtDBQueries.java

```diff
@@ -0,0 +1,121 @@
+            "FROM IDN_AUTH_SESSION_STORE WHERE SESSION_TYPE = ''AppAuthFrameworkSessionContextCache'' " +
+            "AND EXPIRY_TIME > ? AND (TENANT_ID = ? OR TENANT_ID = -1){0} GROUP BY SESSION_ID";
+            // TODO: remove TENANT_ID=-1 from clause when SessionDataStore.removeSessionData() had been updated
+            //       to store session's actual tenant id.
+
+    private static final String LOAD_SESSION_METADATA_DATA = "SELECT SESSION_ID, " +
+            "MAX(CASE WHEN PROPERTY_TYPE = ''IP'' THEN VALUE END) AS IP_ADDRESS, " +
+            "MAX(CASE WHEN PROPERTY_TYPE = ''Login Time'' THEN VALUE END) AS LOGIN_TIME, " +
+            "MAX(CASE WHEN PROPERTY_TYPE = ''Last Access Time'' THEN VALUE END) AS LAST_ACCESS_TIME, " +
+            "MAX(CASE WHEN PROPERTY_TYPE = ''User Agent'' THEN VALUE END) AS USER_AGENT " +
+            "FROM IDN_AUTH_SESSION_META_DATA GROUP BY SESSION_ID";
+
+    private static final String LOAD_APP_DATA = "SELECT ID, APP_NAME, UUID FROM SP_APP {1}";
+
+    private static final String LOAD_SESSION_APP_DATA_MYSQL = "SELECT SESSION_ID, " +
+            "GROUP_CONCAT(CONCAT_WS('':'', APP_ID, APP_NAME, SUBJECT, UUID) SEPARATOR ''|'') AS APPLICATIONS " +
+            "FROM IDN_AUTH_SESSION_APP_INFO ai JOIN APPS a ON a.ID = ai.APP_ID " +
+            "WHERE APP_ID IN (SELECT ID FROM APPS){2} GROUP BY SESSION_ID";
+
+    private static final String LOAD_SESSION_APP_DATA_POSTGRES = "SELECT SESSION_ID, " +
+            "STRING_AGG(CONCAT_WS('':'', APP_ID, APP_NAME, SUBJECT, UUID), ''|'') AS APPLICATIONS " +
+            "FROM IDN_AUTH_SESSION_APP_INFO ai JOIN APPS a ON a.ID = ai.APP_ID " +
+            "WHERE APP_ID IN (SELECT ID FROM APPS){2} GROUP BY SESSION_ID";
+
+    private static final String LOAD_SESSION_APP_DATA_DB2 = "SELECT SESSION_ID, " +
+            "LISTAGG(APP_ID CONCAT '':'' CONCAT APP_NAME CONCAT '':'' CONCAT SUBJECT CONCAT '':'' CONCAT UUID), '|') " +
+            "AS APPLICATIONS FROM IDN_AUTH_SESSION_APP_INFO ai JOIN APPS a ON a.ID = ai.APP_ID " +
+            "WHERE APP_ID IN (SELECT ID FROM APPS){2} GROUP BY SESSION_ID";
+
+    private static final String LOAD_SESSION_APP_DATA_ORACLE = "SELECT SESSION_ID, " +
+            "LISTAGG(APP_ID || '':'' || APP_NAME || '':'' || SUBJECT || '':'' || UUID, ''|'') " +
+            "WITHIN GROUP (ORDER BY APP_ID) AS APPLICATIONS FROM IDN_AUTH_SESSION_APP_INFO ai " +
+            "JOIN APPS a ON a.ID = ai.APP_ID WHERE APP_ID IN (SELECT ID FROM APPS){2} GROUP BY SESSION_ID";
+
+    private static final String LOAD_SESSION_APP_DATA_MSSQL = "SELECT SESSION_ID, " +
+            "STRING_AGG(CONCAT_WS('':'', APP_ID, APP_NAME, SUBJECT, UUID), ''|'') AS APPLICATIONS " +
+            "FROM IDN_AUTH_SESSION_APP_INFO ai JOIN APPS a ON a.ID = ai.APP_ID " +
+            "WHERE APP_ID IN (SELECT ID FROM APPS){2} GROUP BY SESSION_ID";
+
+    public static final String LOAD_SESSIONS_MYSQL = "WITH SESSIONS AS (" + LOAD_SESSION_STORE_DATA + "), " +
```

Comment:
As raised by @vihanga-liyanage 'WITH' keyword will not work in pre mysql 8.0 . Please revise the query to work with mysql versions from 5.7

Reply:
MySQL query updated to work on 5.7